doc/tutorials/specification-language.md

```diff
@@ -0,0 +1,97 @@
+One way to set metadata is the command line tool [kdb meta-set](../help/kdb-meta-set.md) which we have used in the upper example.
+For the [validation plugin](../../src/plugins/validation/README.md) which supports regular expression checks there exists even its own
+command line tool [kdb vset](../help/kdb-vset.md).
+
+If you are writing validation for dozens of configuration settings, the command line might not be the most convenient way. You can also
+write specifications in any file format of your desire. Examples can be seen [here](../../examples/spec). In Elektra, all file formats
+can have metadata but some needed extra tinkering like any `yaml` parser since metadata are not natively supported. We recommend the `ini`
+file format as it natively supports metadata. So the upper example
+could be rewritten in an `ini` file in the following way:
+
+```ini
+[]
+mountpoint = test-specification.ni
+infos/plugins = type
+
+[bool]
+type = boolean
+```
+
+To integrate a configuration into the key database, we mount it by using the property `mountpoint` in the configuration specification.
+This will be mounted into the spec namespace with the following command:
+
+```shell script
+kdb mount </absolute/path/to/ini/file.ini> spec/tests ni
+kdb spec-mount /tests
+```
+
+The `spec-mount` command tells Elektra to load all relevant plugins. It is smart enough to automatically detect the needed plugins
+by looking at all metadata. You might need to remove some metadata to not load certain plugins that cause the exceeding of the maximum
+plugin number.
+
+What remains to know are the metadata names and meanings. We have used the `type` plugin which provides the metadata `type` metadata along with
+many other metadata such as `check/enum`. You can read all available metadata in the plugin READMEs which all have to have a
+so called `contract` at the beginning of each file. This contract also contains the `infos/metadata` data which lists all available
+metadatas. We recommend to look through those plugins READMEs to know how they are intended to use.
+
+### Nice to know
+
+Many plugins write metadata with arrays in them such as the `type` plugin which allows to list all enumerations for configuration settings.
+We have a nice [tutorial](../tutorials/arrays.md) for you to get used to arrays.
+
+You can also set validation for multiple configurations settings at once by using [globbing](../../src/plugins/glob/README.md).
+
+If you decide to write metadata via a file (and especially the `ini` format) you should be aware of multiline strings that can cause
+parsing issues. Descriptions of metadata often take up multiple lines and
+there are multiple `ini` format standards which tell you differently how to write them.
+Elektra also has two ini plugins (`ni` and `ini`) which also handle multilines differently. We recommend the `ni` plugin which also
+works correctly with our provided [examples](../../examples/spec).
```

Comment:
You did not mention doc/METADATA.md? Please explain how to read and use this file. This is essential for understanding the specification language.

Reply:
> Please explain how to read and use this file.

I guess you mean metadata. *ini*? I added a paragraph concerning that, I hope it correctly reflects what you have in mind

Reply:
Yes, METADATA.ini. I am not sure if a paragraph is enough to understand what this file is about and how to practically use it. Understanding METADATA.ini is essential for using the specification language.

Reply:
> Understanding METADATA.ini is essential for using the specification language.

I actually never have used this file and are confused why it is so important. I think you will need to explain it to me first so I can add it to the tutorial

Reply:
It described formally and in detail in book.libelektra.org

In essence METADATA.ini describes the semantic of the specification language.

Reply:
Where exactly you write about this file in the book? I have read through `Abstraction over Embedded and Separated Metadata` but do not think that this helps in this case

Reply:
page 92ff

Reply:
> page 92ff

I have read through several pages and understand that it is a meta-meta-specification. I though am not sure how to practically use this file. The actually most useful information for me is in the METADATA.ini file itself which tells that you can describe which configuration files in which formats are present on the system. I also assume that one can change what is regarded as a comment symbol but I am not sure what exactly it can achieve otherwise.

Reply:
Would a small hint to the book be interesting here? If yes, then we really need to specify which sites are relevant. I would not have found it without the hint to 92ff.

Reply:
Currently, it is meant to be read from beginning to end. I plan to rewrite the book to a scriptum so that is more structured by topics but it will need at least one more year to finish this.